GENOTYPE = config['GENOTYPE']
ASSEMBLY_FASTA = config['ASSEMBLY_FASTA']
TREP_FASTA = config['TREP_FASTA']
SUBGENOMES = config['SUBGENOMES']

rule all:
    input:
        expand("results/{genotype}.masked.fasta", genotype = GENOTYPE), 
        expand("results/{genotype}.te.gff", genotype = GENOTYPE)

rule subset_subgenomes:
    input:
        ASSEMBLY_FASTA
    output:
        "results/subgenomes/{subgenome}.fa"
    log:
        "logs/subset_{subgenome}.log"
    shell:
        "seqkit grep -rp '{wildcards.subgenome}' {input} > {output} 2> {log}"

rule edta:
    input:
        genome = "results/subgenomes/{subgenome}.fa", 
        trep = TREP_FASTA
    output:
        "results/edta/{subgenome}.mod.EDTA.TElib.novel.fa"
    params:
        anno = "0"
    log:
        "logs/edta/{subgenome}.log"
    threads: 24
    shell:
        "perl /project/gbru_wheat2/conda/edta_env/bin/EDTA.pl --genome {input.genome} --curatedlib {input.trep} --anno {params.anno} --threads {threads} 2> {log}"

rule mask:
    input:
    output:
    log:
    threads:
    shell:
        "perl /project/gbru_wheat2/conda/edta_env/bin/make_masked.pl -genome unmasked_genome.fa [options] -rmout [file] Required. The repeatmasker.out file"
